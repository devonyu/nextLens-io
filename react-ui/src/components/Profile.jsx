import React, { Component } from 'react';
import { Button, Card, Icon, Image } from 'semantic-ui-react';

export default class Profile extends Component {
    render() {
        return(
                <Card>
                    <Image src={this.props.userInformation.profileimgurl} />
                    <Card.Content>
                    <Card.Header>{this.props.userInformation.firstname}</Card.Header>
                    <Card.Meta>Current Mount: {this.props.userInformation.mount}</Card.Meta>
                    <Card.Description>
                    {this.props.userInformation.about}
                    <br/>
                    <Icon name='camera retro' />
                    Owns X lenses
                    </Card.Description>
                    </Card.Content>
                    <Card.Content extra>
                    <Button onClick={()=>{console.log('update profile handler')}} size='small'>Edit Profile</Button>
                    </Card.Content>
                </Card>
        )
    }
}