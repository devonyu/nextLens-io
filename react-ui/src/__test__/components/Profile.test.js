import React from 'react';
import Profile from '../../components/Profile';
import renderer from 'react-test-renderer';

test('Loads Profile Card with Props', () => {
  const userInformation = {
    firstname: 'Tester',
    profileimgurl: 'https://devonyu.com/pf2.png',
    mount: '3',
    email: 'devonyu415@gmail.com'
  };
  const component = renderer.create(<Profile userInformation={userInformation} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
