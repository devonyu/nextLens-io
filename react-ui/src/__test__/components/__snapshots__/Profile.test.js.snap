// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loads Profile Card with Props 1`] = `
<div
  className="ui card"
  onClick={[Function]}
>
  <img
    className="ui image"
    onError={[Function]}
    src="test-file-stub"
  />
  <div
    className="content"
  >
    <div
      className="header"
    >
      Tester
    </div>
    <div
      className="meta"
    />
  </div>
</div>
`;
